packages/editor/src/components/use-styles/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+export function useStyles() {
+	const { styles, isReady } = useSelect( ( select ) => {
+		const { getEditedEntityRecord, hasFinishedResolution } =
+			select( coreStore );
+		const _globalStylesId =
+			select( coreStore ).__experimentalGetCurrentGlobalStylesId();
+		const record = _globalStylesId
+			? getEditedEntityRecord( 'root', 'globalStyles', _globalStylesId )
+			: undefined;
+
+		let hasResolved = false;
+		if (
+			hasFinishedResolution( '__experimentalGetCurrentGlobalStylesId' )
+		) {
+			hasResolved = _globalStylesId
+				? hasFinishedResolution( 'getEditedEntityRecord', [
+						'root',
+						'globalStyles',
+						_globalStylesId,
+				  ] )
+				: true;
+		}
+
+		return {
+			isReady: hasResolved,
+			styles: record?.styles,
+		};
+	}, [] );
+
+	// Make sure to recompute the styles when hasResolved changes.
+	const config = useMemo( () => {
```

Comment:
I was wondering whether we need `useMemo` on the return value of `useSelect` here since I assumed `useSelect` does the shallow comparison on the return values?

cc @jsnajdr for head check.

Reply:
The function of this `useMemo` is to keep an unchanged `config` object when its fields, `styles` and `isReady` are unchanged. So that `useStyles` doesn't return a different object on each call.

But the way how `useStyles` is used makes this unnecessary. The return value is destructured:
```js
const { styles } = useStyles();
```
so the `config` object disappears, it's not used anywhere as a prop or an effect dependency.

There is this comment above the `useMemo` call:
> Make sure to recompute the styles when hasResolved changes.

It's confusing. The `styles` are fully computed, the hook is not recomputing anything. It's just ensuring that referentially-identical input leads to referentially-identical output.